### DISCUSSION TITLE
Increase frequency of the stale bot checks

### DISCUSSION THREAD

**Discussion by AndrewSav** [hairyhenderson/gomplate#1773] [General]
Continued from [here](https://github.com/hairyhenderson/gomplate/issues/862#issuecomment-1606209065).

On most public repo I'm following the stale bot is run about every 5 minutes. It means that if it marks a conversation as stale and a person go and contribute to the thread they can soon see the issue is unmarked as stale. This shows a modicum of respect for people's time: if they are made to do work to keep the issue open at least they do not need to check manually in a week time that the issue remains non-stale.

I'm in general against the stale bot - all it causes is a stream of "+1"s, "I need this too"s and "Still not working"s for years. But I do understand and respect that some repo maintainers choose to have them.

IMO it's healthy for the community to have sane settings, when we do have those, and having the check once a week is very non-standard. [As we discovered](https://github.com/hairyhenderson/gomplate/issues/862#issuecomment-1606209065) there is no monetary incentive to keep it long it either. The way I see it:

- The downside of changing to a shorter cycle: none 
- The upside of changing to a shorter cycle: contributors are not forced to manually check in a week's time if the issue lost the "Stale" status. Also - everyone else is doing that, so there is nothing controversial about it either.

Note, that the stale bot _does miss a beat sometimes_. From experience, I saw occurrences, when a message was not accounted for and a second message was required to remove the stale label. So one _does_ have to check if they want to make sure that the "Stale" label is gone.

**Comment by hairyhenderson** [comment]
Fair. See also https://github.com/hairyhenderson/gomplate/discussions/1712 for another dissenting opinion, and my response. I'll try not to repeat the points I made there.

To explain something I said in the [original conversation](https://github.com/hairyhenderson/gomplate/issues/862#issuecomment-1606209065):

> In that case, I could run that workflow more often. However, I still don't see a need.

By "I still don't see a need", I meant that there is no strict technical reason to run it more often. The periods between inactivity-to-stale and stale-to-closed are relatively long, and running the workflow once a week gives more than enough time for activity to happen and the workflow to detect them.

I understand that it might feel like commenting on issues has no effect, and I believe you when you say you've seen the workflow missing updates. That's reason enough for increasing the frequency, and so I will. My misunderstanding on billing complicated matters unnecessarily, and I apologize for that.

I've made another (and IMO more useful) change already: I've configured the stale workflow to ignore issues and PRs that are assigned to milestones. In assigning an issue to a milestone, I'm effectively saying "yes, this is planned, so nobody needs to post keepalive comments".

All this being said, please continue to participate in this project. I find your participation valuable and insightful. So even though I can be slow to respond, and outright wrong at times, don't let that alienate you.

**Reply by AndrewSav** [reply]
>  That's reason enough for increasing the frequency, and so I will. 

Thank you.

> All this being said, please continue to participate in this project. I find your participation valuable and insightful. So even though I can be slow to respond, and outright wrong at times, don't let that alienate you.

Really appreciate you saying this, and your measured response. Apologies for losing my temper.
